### ISSUE TITLE
Auto convert markdown syntax during import 

### ISSUE THREAD

**Issue by sinsiynan** [siyuan-note/siyuan#8293]
### In what scenarios do you need this feature?

As the title states, this feature is for cases where a large document is imported to Siyuan with a foreign syntax. 

The end goal is that the internal references in the source document, which could be innumerable, is transferred to Siyuan content block ID's, or virtual references (a feature of Siyuan)

To be fair, I noticed a setting "Virtual reference" in Siyuan which I enabled. In the TOC (table of contents) in my large document  -  each entry has brackets [TOC-title] and refers to headings in 300 page document. Almost non of the virtual references is detected correctly and I do not understand how the inclusion and exclusion lists work in this regard. 

### Describe the optimal solution

By no means am I a developer as talented others here, but I have conceptual insight (user perspective)

**The feature would be:**
1. During import of a large document (either markdown, html or else) internal referencing/linking is lost. It does not correspond to Siyuan content block ID, but preferably, the referencing is converted to Siyuan's own content block ID.   
2. During import the syntax can be markdown or else, so the end user should be prompted to adjust  & select syntax from the source document to tweak the _detection rules_ for output in the destination document.

**To explain bullet number # 2 more detailed:**
The source document in case of markdown can be basic markdown, extended markdown or else. The source could be other formats as well (html, .docx, etc.)
Again, during import Syntax detection rules should be adjustable by user for foreign unknown syntax or to force detection rules for individual instances. 

### Describe the candidate solution

For example, during import user is presented with empty fields that looks like: ___    =    ___. 
A user can then define detection rules and add more fields:  
[]    =    (()) . (basic markdown)

### Other information

A correct working virtual reference would, if working correctly for me,  negate this feature sufficiently, but in the Siyuan User Guide I miss concrete examples of how the keyword inclusion/exclusion lists work. 

Basically, in short, i wish the virtual reference had a "force syntax detection" rules definable in inclusion and exclusion lists. 
Keywords are: syntax converter, force detection rules, user definable, import enhancement

**Comment by 88250** [comment]
Hello, we have tried a similar idea before. At that time, we planned to implement the reference syntax for importing Obsidian, but it was very difficult. The main reason is that these syntaxes are ambiguous, so there is no way to convert them completely correctly. Later, we can only abandoned https://github.com/siyuan-note/siyuan/issues/3557 (in Chinese)

Similarly, for more file format (or syntax) conversion support, we suggest to handle it externally. SiYuan can only provide "standard" Markdown import support. Only in this way can we avoid introducing more problems as much as possible.

Regarding the virtual reference, it is not persisted in the data file, but is calculated and rendered at runtime, so it has nothing to do with the import conversion format. The rules of virtual references are indeed somewhat complicated. Its main principle is based on search. You can search for some items of concern in Settings-Search-Virtual References, enable and configure the inclusion/exclusion list in Settings-Editor-Virtual References, you can try enabling virtual references to see how the user guide changes, for example:

![image](https://github.com/siyuan-note/siyuan/assets/873584/127332fc-b00e-4a5d-83ad-b86ec6fa38ab)
